agent/src/softwarecontaineragent.h

```diff
@@ -270,6 +273,11 @@ class SoftwareContainerAgent
      * Responsible for parsing any dynamic values given when creating a container
      */
     ContainerOptionParser m_optionParser;
+
+    /*
+     * Responsible to provide connection to containers.
```

Comment:
I think this could be described as the factory. A factory is a common pattern and further details should be in that class. This comment is a bit confusing at the moment.